frontend/src/pages/Compare.tsx

```diff
@@ -212,11 +214,13 @@ class Compare extends Page<{}, CompareState> {
       return;
     }
 
+    const selectedIds = runs.map(r => r.run!.id!);
     this.setStateSafe({
-      runs,	
-      selectedIds: runs.map(r => r.run!.id!),	
-      workflowObjects,	
-    }, () => this._loadParameters());
+      runs,
+      selectedIds,
+      workflowObjects,
+    });
+    this._loadParameters(selectedIds);
```

Comment:
You mean consolidating `setState` calls?
I think the main issue isn't how many times the container component renders though, it's cascading the rendering. Components should be pure wherever they can be, so that a parent re-render does not trigger all of its subtree to re-render as well.

Reply:
That, and removing unnecessary `setState` callbacks that could be achieved using parameters